cmd/namespace-lock.go

```diff
@@ -137,8 +133,14 @@ func (n *nsLockMap) lock(volume, path string, lockSource, opsID string, readLock
 
 	// Locking here can block.
 	if readLock {
+		if globalFSMutex != nil {
```

Comment:
The namespace lock should be taken before the shared filesystem lock is attempted. This is the only we can prevent races between two request handlers attempting a lock on the same file from the same minio server process.